src/fastqsplitter/split.pyx

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# cython: language_level=3
+
+
+import io
+from typing import List
+
+
+def filesplitter(input_handle: io.BufferedReader,
+                 output_handles: List[io.BufferedWriter],
+                 lines_per_block=100):
```

Comment:
A single fastq record is at least four lines, maybe I'm overlooking something but if I were to set this to some value which is not divisible by 4, I would end up with invalid fastq files, right?

Maybe one more check to make sure lines_per_block is divisible by 4 should be added. (And a comment stating that multi-line fastq records are not supported)

Reply:
This is a filesplitter. Thus it is not aware of the fastq format.
The fastq format is handled upstream. Where the group_size is multiplied by  4. This results in a number that is always divisible by 4.

Reply:
https://github.com/LUMC/fastqsplitter/blob/def0c115e1165e79e71856672e1f9515095ede7b/src/fastqsplitter/__init__.py#L108

Reply:
> This application does NOT work with multiline fastq sequences.

Is stated in the README.

Reply:
:+1: nevermind :stuck_out_tongue: